internal/testhelper/wftest/proxy.go

```diff
@@ -0,0 +1,35 @@
+package wftest
+
+import (
+	"github.com/wavefrontHQ/wavefront-operator-for-kubernetes/internal/util"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Proxy(options ...func(*appsv1.Deployment)) *appsv1.Deployment {
+	proxy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.ProxyName,
+			Namespace: DefaultNamespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name":      "wavefront",
+				"app.kubernetes.io/component": "proxy",
+			},
+		},
+		Status: appsv1.DeploymentStatus{
+			AvailableReplicas: 1,
+			Replicas:          1,
+		},
+	}
+	for _, option := range options {
+		option(proxy)
+	}
+	return proxy
+}
+
+func WithReplicas(availableReplicas, replicas int) func(*appsv1.Deployment) {
```

Comment:
I like where you are going with this - sending the customizations we might want to make as functions.

Reply:
In an earlier attempt, I went crazy with this idea (customization functions for everything). However, the power-to-weight ratio seemed pretty low since many customizations were only used once. However, replicas was used enough that I felt like it was worth its own helper.

Reply:
I learned a lot of good practices and interesting implementation decisions from reviewing this PR. Thank you @m25n!